drivers/hsc/hsc_saul.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2020 Deutsches Zentrum für Luft- und Raumfahrt e.V. (DLR)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_hsc
+ * @{
+ *
+ * @file
+ * @brief       SAUL adaption for HSC device
+ *
+ * @author      Quang Pham <phhr_quang@live.com>
+ * @}
+ */
+
+#include <string.h>
+#include "saul.h"
+#include "hsc.h"
+
+static int read_temperature(const void *dev, phydat_t *res)
+{
+    res->val[0] = hsc_read_temperature(dev);
+    res->unit = UNIT_TEMP_C;
+    res->scale = -1;
+    return 1;
+}
+
+static int read_pressure(const void *dev, phydat_t *res)
+{
+    res->val[0] = (int16_t)(hsc_read_pressure(dev));
+    res->unit = UNIT_BAR;
+    res->scale = -6;
```

Comment:
```suggestion
    res->unit = UNIT_BAR;
    res->scale = -6;
    int32_t value = hsc_read_pressure(dev);
    phydat_fit(res, &value, 1);
    return 1;
```

When you just cast an `int32_t` to an `int16_t`, data above 32.767 mBar will be cropped. However, `phydat_fit()` will instead adjust `res->scale` until the value in fits into 16 bit. It even does correct scientific rounding.